energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -179,7 +179,14 @@ pub trait WeeklyRewardsSplittingModule:
             Energy::default()
         };
 
-        let prev_week = current_week - 1;
+        // Update Current/Previous Week
+        if current_week != self.current_active_week().get() {
```

Comment:
Maybe use the mapper like this:
`let current_global_active_week_mapper = self.current_active_week();`
Then you can only use
`current_active_week_mapper.get()` and `current_active_week_mapper.set()` without instantiating the mapper every time.

Reply:
Nice catch, thanks!